### ISSUE TITLE
Adding AlertsProvider adds vertical space

### ISSUE THREAD

**Issue by rchavik** [kuasha420/react-native-paper-alerts#1]
When I add `<AlertsProvider>`, it adds empty space below the application. Traced it to the `marginVertical: -300` in https://github.com/kuasha420/react-native-paper-alerts/blob/906de4cbd500fd4d1e71a43d0ddc839e811d25d8/src/provider.tsx#L474

Commenting it out seems to remove the weird space.

Is there a way to customize these styles or is this purely intended for internal use?

**Comment by kuasha420** [comment]
Can you give me a screenshot? I think I put the style to debug/test something then forgot to remove before releasing. F.

**Comment by rchavik** [comment]
Normally the app has no scrollbar. `<AlertsProvider>` seems to add space causing the page to be scrollable.

![image](https://user-images.githubusercontent.com/39490/128978538-31b2a2b3-6bcb-46aa-80ad-2b214238944d.png)

Scrolling down just reveals an empty space:
![image](https://user-images.githubusercontent.com/39490/128978802-1d960437-4308-4c5a-b443-156f93c4d4ab.png)

**Comment by kuasha420** [comment]
Thanks. I'll try to reproduce it. I now remember why the margin is there and it's to avoid peaking throw the modal when keyboard is on/off screen. Can you share where you've put the alert provider? Is should be inside SafeAReaProvider and PaperProvider but before anything that has styles. See https://github.com/kuasha420/react-native-paper-alerts/blob/906de4cbd500fd4d1e71a43d0ddc839e811d25d8/example/src/index.tsx#L103

**Comment by rchavik** [comment]
> Thanks. I'll try to reproduce it. I now remember why the margin is there and it's to avoid peaking throw the modal when keyboard is on/off screen. Can you share where you've put the alert provider? 

Here it is: 
- `<Provider>` is from `mobx-react`;
- `<AlertInjector>` is a custom FC that call `useAlerts` and set it up in my mobx store. 
- `<AppRoutes>` is my navigation routes, using `react-navigation`, stack, etc.

```jsx
    <Provider { ...stores }>
        <PaperProvider theme={ theme }>
            <AlertsProvider>
                <AlertInjector store={ stores.store }>
                    <AppRoutes>
                        { props.children }
                    </AppRoutes>
                </AlertInjector>
            </AlertsProvider>
        </PaperProvider>
    </Provider>
```

I didn't have the `SafeAreaProvider`, but after adding it in, still no effect. 

Re: if the margin is added to compensate for keyboard on/off, would using this https://reactnative.dev/docs/keyboardavoidingview help?

**Comment by kuasha420** [comment]
Very weird, the example project also has similar setup to yours and it doesn't exhibit the same issue. 

The margin is here to prevent the white flash behind the keyboard when it's being dismissed as seen in the video. KeyboardAvoidingView doesn't work here sadly. Let me know if you have any solution to this. 


https://user-images.githubusercontent.com/6187401/128991821-572e4551-45cb-4171-9cad-a9cc37492a0b.mp4

**Comment by rchavik** [comment]
> Thanks. I'll try to reproduce it. 

Can you try to repro using https://github.com/rchavik/deep-linking-example (branch `repro-1`)

I even tried removing `NavigationContainer`, but the margin remained:

```diff
@@ -169,9 +168,7 @@ export default function App() {
     <SafeAreaProvider initialMetrics={ initialWindowMetrics }>
       <Provider>
         <AlertsProvider>
-          <NavigationContainer linking={ linking }>
-            <SidebarLeft />
-          </NavigationContainer>
+          <TabB />
         </AlertsProvider>
       </Provider>
     </SafeAreaProvider>
```

**Comment by rchavik** [comment]
> Can you try to repro using https://github.com/rchavik/deep-linking-example (branch `repro-1`)
> 
> I even tried removing `NavigationContainer`, but the margin remained:

This is in branch `repro-2` with most things related to navigation have been removed. So it does not seem due to combination `react-navigation`.

**Comment by kuasha420** [comment]
Hey, thanks for the investigation and pr. Sadly the PR adds the "see through the modal background" issue pointed in the video as seen here. 

![screenshot-2021-08-11_18 20 09 398](https://user-images.githubusercontent.com/6187401/129027734-cf629b91-4304-4306-b61c-6edb400204ac.png)

But your PR has given me an idea and I'm going to see of that fixes the issue you described with expo web without breaking the android looks. 

Also, I figured why the web example in this project (`yarn example web`) is working fine, it's because of the additional CSS applied in the `index.html` file here: 

https://github.com/kuasha420/react-native-paper-alerts/blob/906de4cbd500fd4d1e71a43d0ddc839e811d25d8/example/web/index.html#L24

However, expo web is clearly important and needs fixing. 
Stay tuned ...

**Comment by mikehardy** [comment]
I just integrated this package to solve the "react-native-web doesn't implement the Alert API" issue, thanks for this!
I applied the style you mentioned: https://github.com/kuasha420/react-native-paper-alerts/blob/906de4cbd500fd4d1e71a43d0ddc839e811d25d8/example/web/index.html#L24
I'm not doing keyboard prompts, just standard text + button alerts, and it seems to be working great, for what that's worth
https://github.com/invertase/react-native-firebase-authentication-example
https://invertase.github.io/react-native-firebase-authentication-example/

**Comment by adilusman51** [comment]
@kuasha420 First of all, thanks a lot for the awesome work on this repository.

I am having the same issue on Expo Web, adding `<AlertsProvider>` adds vertical margin and makes page scrollable with a white space at the bottom (looks even more ugly when Dark theme is applied)

Upon manually removing `marginVertical: -300` in the modules folder, fixes the white space scrolling issue.

Kindly, provide an update on this. 

Suggestion: Remove margin for Simple Alerts (title, text + buttons) and apply vertical margin to Prompts only

**Comment by adilusman51** [comment]
@kuasha420 Another suggestion is not to apply style on Platform.OS === 'web'

**Comment by mikehardy** [comment]
forgive me, but I'm only commenting to give you a huge +1 for the investigation and posting a PR. Open Source :trophy: :smile:

**Comment by kuasha420** [comment]
@adilusman51 your suggestion and PR sounds reasonable and doesn't introduce the see through issues. I've merged it and released 0.1.2. Can you confirm if it fixes the issue? Cheers.

**Comment by adilusman51** [comment]
@kuasha420 Thanks a lot for reviewing the PR. I have verified this issue on the project I was facing this issue and can verify that this  issue is resolved. Kindly, close this issue.

@mikehardy Thanks a lot for your support. Much appreciated. Open Source for the Win 🏆 😄
